client/my-sites/plans/jetpack-plans/product-store/hooks/use-item-lightbox.ts

```diff
@@ -1,22 +1,35 @@
 import { useCallback, useMemo, useState } from 'react';
 import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import { EXTERNAL_PRODUCTS_LIST } from '../../constants';
+import slugToSelectorProduct from '../../slug-to-selector-product';
 import { SelectorProduct } from '../../types';
 
 export const useItemLightbox = () => {
-	const [ currentItem, setCurrentItem ] = useState< SelectorProduct | null >( null );
+	const item = slugToSelectorProduct( window.location.hash.substring( 1 ) );
```

Comment:
Can we sanitize the hash before using it or ensure that only the acceptable values are handled?

We can do so by something like this:

```ts
const sanitizeProductSlug = ( productSlug: string ) => {
	// product slugs can contain only english alphanumeric characters and underscores
	return productSlug.replace( /[^a-z0-9_]/gi, '' );
};
```

Reply:
@manzoorwanijk thanks for spotting this one.  could have been a security issue. 🙇 

I decided to use a more generic `sanitizeLocationHash` name for this so we can reuse it not only for product slugs but any other location hash value. the rules are reasonable enough for generic use case.